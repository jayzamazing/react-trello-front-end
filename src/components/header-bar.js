import React from 'react';
import {connect} from 'react-redux';
import {setCurrentUser, setAuthToken} from '../actions/auth';
import {clearAuthToken} from '../local-storage';
import './header.css';
import logo from '../img/trello-logo-white.svg';
import {Link} from 'react-router-dom';

export class HeaderBar extends React.Component {
    logOut() {
        this.props.dispatch(setCurrentUser(null));
        this.props.dispatch(setAuthToken(null));
        clearAuthToken();
    }
    render() {
        // Only render the log out button if we are logged in
        let buttons;
        if (this.props.loggedIn) {
          buttons = (
            <div className="nav-right">
              <button onClick={() => this.logOut()}>Log out</button>
            </div>
          );
        } else {
          buttons = (
            <div className="nav-right">
              <Link to="/login">
                <button className="btn login-btn">Log In</button>
              </Link>
              <Link to="/register">
              <button className="btn sign-up-btn">Sign Up</button>
              </Link>
            </div>
          );
        }
        return (
            <header className="container-fluid header-bar">
              <nav className="row home-nav">
                <div className="nav-left">
                  <Link to="/" className="logo">
                    <img src={logo} alt="Logo"></img>
                  </Link>
                </div>
                  {buttons}
              </nav>
            </header>
        );
    }
}
const mapStateToProps = state => ({
    loggedIn: state.auth.currentUser !== null
});
export default connect(mapStateToProps)(HeaderBar);
