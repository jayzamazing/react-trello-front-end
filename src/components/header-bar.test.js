'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {HeaderBar} from './header-bar';

describe('header-bar', () => {
  it('should render with login button', () => {
    let location = {
      pathname: '/'
    }
    const wrapper = shallow(<HeaderBar location={location}/>);
    expect(wrapper.find('.login-btn')).toHaveLength(1);
  });
  it('snapshot with login button', () => {
    let location = {
      pathname: '/boards'
    }
    const wrapper = shallow(<HeaderBar location={location}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render with logout button', () => {
    let location = {
      pathname: '/'
    }
    const wrapper = shallow(<HeaderBar location={location} loggedIn="true"/>);
    expect(wrapper.find('.logout-btn')).toHaveLength(1);
  });
  it('snapshot with logout button', () => {
    let location = {
      pathname: '/boards'
    }
    const wrapper = shallow(<HeaderBar location={location} loggedIn="true"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //ensure that header cannot be found when it is rendered
  it('should not be visible', () => {
    let location = {
      pathname: '/register'
    }
    let wrapper = shallow(<HeaderBar location={location}/>);
    expect(wrapper.find('header')).toHaveLength(0);
    location = {
      pathname: '/login'
    }
    wrapper = shallow(<HeaderBar location={location}/>);
    expect(wrapper.find('header')).toHaveLength(0);
  });
  //ensure the snapshot has no data for either of the following paths
  it('should not be visible', () => {
    let location = {
      pathname: '/register'
    }
    let wrapper = shallow(<HeaderBar location={location}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
    location = {
      pathname: '/login'
    }
    wrapper = shallow(<HeaderBar location={location}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //check dispatch is fired off when logout button is clicked
  it('dispatches logout', () => {
    let location = {
      pathname: '/boards'
    }
    const dispatch = jest.fn();
    const wrapper = shallow(<HeaderBar location={location} dispatch={dispatch} loggedIn="true"/>);
    wrapper.find('.logout-btn').at(0).simulate('click');
    expect(dispatch).toHaveBeenCalledWith({"authToken": null, "type": "SET_AUTH_TOKEN"});
    expect(dispatch).toHaveBeenCalledWith({"currentUser": null, "type": "SET_CURRENT_USER"});
  });
});
