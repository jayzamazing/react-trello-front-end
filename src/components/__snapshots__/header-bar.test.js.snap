// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header-bar should not be visible 1`] = `""`;

exports[`header-bar should not be visible 2`] = `""`;

exports[`header-bar snapshot with login button 1`] = `
<header
  className="container-fluid header-bar"
>
  <nav
    className="row home-nav"
  >
    <div
      className="nav-left"
    >
      <Link
        className="logo"
        replace={false}
        to="/"
      >
        <img
          alt="Logo"
          src="trello-logo-white.svg"
        />
      </Link>
    </div>
    <div
      className="nav-right"
    >
      <Link
        replace={false}
        to="/login"
      >
        <button
          className="btn login-btn"
        >
          Log In
        </button>
      </Link>
      <Link
        replace={false}
        to="/register"
      >
        <button
          className="btn sign-up-btn"
        >
          Sign Up
        </button>
      </Link>
    </div>
  </nav>
</header>
`;

exports[`header-bar snapshot with logout button 1`] = `
<header
  className="container-fluid header-bar"
>
  <nav
    className="row home-nav"
  >
    <div
      className="nav-left"
    >
      <Link
        className="logo"
        replace={false}
        to="/"
      >
        <img
          alt="Logo"
          src="trello-logo-white.svg"
        />
      </Link>
    </div>
    <div
      className="nav-right"
    >
      <button
        className="btn logout-btn"
        onClick={[Function]}
      >
        Log out
      </button>
    </div>
  </nav>
</header>
`;
